synapse/logging/opentracing.py

```diff
@@ -236,8 +236,16 @@ class _DummyTagNames:
 try:
     from rust_python_jaeger_reporter import Reporter
 
+    # jaeger-client 4.7.0 requires that reporters inherit from BaseReporter, which
+    # didn't exist before that version.
```

Comment:
Ahh, I didn't realise this was new in 4.7.

(Is the inheritance you describe a runtime requirement, or just part of the type annotations added in 4.7?)

Reply:
Based on jaegertracing/jaeger-client-python#329 I think it is really just there for type-annotations.

It does seem that they consolidated some logic into it. (The `close` method as you noticed.) But it seems to essentially be a no-op so I don't think this is huge.